### PR TITLE
Fix for ThrowInt32OverflowException in Issue.Events.GetAllForRepository

### LINKED ISSUE #2031: ThrowInt32OverflowException on Issue.Events.GetAllForRepository
Using octokit 0.36.0

The following code causes an exception:

```csharp
var results = await client.Issue.Events.GetAllForRepository("dotnet", "docs");
```

```
This exception was originally thrown at this call stack:
	System.Convert.ThrowInt32OverflowException()
	System.Convert.ToInt32(long)
	long.System.IConvertible.ToInt32(System.IFormatProvider)
	System.Convert.ChangeType(object, System.Type, System.IFormatProvider)
	Octokit.PocoJsonSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJson.cs
	Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJsonSerializer.cs
	Octokit.PocoJsonSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJson.cs
	Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJsonSerializer.cs
	Octokit.PocoJsonSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJson.cs
	Octokit.Internal.SimpleJsonSerializer.GitHubSerializerStrategy.DeserializeObject(object, System.Type) in SimpleJsonSerializer.cs
	...
	[Call Stack Truncated]
```

### PR DESCRIPTION
**Pull request by terrance00**
Fixes #2031

 - Updated Model with long Id
 - Fixed tests
 - Updated IssuesClient and interface
 - Updated IssuesEventsClient and interface

- Double checked that the same tests still failing for the same reasons.

### CODE REVIEW

**Comment by shiftkey** [comment]
> - Double checked that the same tests still failing for the same reasons.

Which test is this? And is it supposed to fail?

**Comment by terrance00** [comment]
Four instances of:
```
CanPopulateObjectFromSerializedData

``` 
Unless I missed something at Dev startup - but directly from master - the four of those fail.

Also there are some integration tests that fail...

**Comment by shiftkey** [comment]
> Unless I missed something at Dev startup - but directly from master - the four of those fail.

It's weird because both [master](https://ci.appveyor.com/project/github-windows/octokit-net/builds/28275105) and [this PR](https://ci.appveyor.com/project/github-windows/octokit-net/builds/28356883) don't seem to have issues with those tests.

These are the usages as I understand it, and I can't spot how the changes relate to your PR:
https://github.com/octokit/octokit.net/search?utf8=%E2%9C%93&q=CanPopulateObjectFromSerializedData&type=

**Comment by shiftkey** [comment]
> Also there are some integration tests that fail - but I assumed that there are some environment variables I don't have.

There's some setup required - https://github.com/octokit/octokit.net/blob/master/CONTRIBUTING.md#slow-tests but I'm not worried about them for this PR.

**Comment by terrance00** [comment]
Yeah it is those four, they do break for me on master as well - without any changes. Any idea what to try?

I get this:
```
System.Runtime.Serialization.SerializationException : Unable to find assembly 'Octokit, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
```

**Comment by terrance00** [comment]
Seems to be happening in:
```
Exception thrown: 'System.Runtime.Serialization.SerializationException' in System.Runtime.Serialization.Formatters.dll
Exception thrown: 'System.Reflection.TargetInvocationException' in System.Private.CoreLib.dll
```

I wonder if it is dotnet core 3 related...

**Comment by shiftkey** [comment]
@terrance00 please open a new issue with details about it - we run the tests for `netcoreapp2.0` and `net452` so I'm curious which is failing for you.

**Comment by terrance00** [comment]
Hey man, sorry for the delay.

I have resigned to running the tests in the command line - and that does not fail for those tests.

Leads me to believe that it must be something IDE related in relation the `NO_SERIALIZABLE` if clause.

File: Octokit.Reactive/Clients/IObservableIssuesClient.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public interface IObservableIssuesClient
         /// <param name="number">The issue number</param>
         [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", MessageId = "Get",
              Justification = "Method makes a network request")]
-        IObservable<Issue> Get(string owner, string name, int number);
```
Comment (shiftkey) [inline Octokit.Reactive/Clients/IObservableIssuesClient.cs]: The `number` used in this interface is not the one overflowing. Can we undo the changes in this file to simplify the diff?

File: Octokit/Clients/IIssuesClient.cs [hunk 2]
```diff
@@ -52,7 +52,7 @@ public interface IIssuesClient
         /// <param name="number">The issue number</param>
         [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", MessageId = "Get",
              Justification = "Method makes a network request")]
-        Task<Issue> Get(string owner, string name, int number);
+        Task<Issue> Get(string owner, string name, long number);
```
Comment (shiftkey) [inline Octokit/Clients/IIssuesClient.cs]: Same thing here - this `number` used in this interface is not the one overflowing. Can we undo the changes in this file to simplify the diff?

File: Octokit.Reactive/Clients/IObservableIssuesEventsClient.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ public interface IObservableIssuesEventsClient
         /// <param name="owner">The owner of the repository</param>
         /// <param name="name">The name of the repository</param>
         /// <param name="number">The issue number</param>
-        IObservable<EventInfo> GetAllForIssue(string owner, string name, int number);
+        IObservable<EventInfo> GetAllForIssue(string owner, string name, long number);
```
Comment (shiftkey) [inline Octokit.Reactive/Clients/IObservableIssuesEventsClient.cs]: Same thing for this file - `number` here is the issue number itself, no the `EventInfo` number that is being deserialized.

File: Octokit/Clients/IIssuesEventsClient.cs [hunk 4]
```diff
@@ -21,7 +21,7 @@ public interface IIssuesEventsClient
         /// <param name="owner">The owner of the repository</param>
         /// <param name="name">The name of the repository</param>
         /// <param name="number">The issue number</param>
-        Task<IReadOnlyList<EventInfo>> GetAllForIssue(string owner, string name, int number);
+        Task<IReadOnlyList<EventInfo>> GetAllForIssue(string owner, string name, long number);
```
Comment (shiftkey) [inline Octokit/Clients/IIssuesEventsClient.cs]: Same thing for this file - `number` here is the issue number itself, no the `EventInfo` `number` that is being deserialized.

**Comment by shiftkey** [comment]
Thanks for taking a shot at this @terrance00!

I received #2060 which was the minimal fix so this can be closed out, but I hope that you can find time in the future for another contribution.
